main.tf

```
@@ -34,9 +34,10 @@ locals {
   certificate_arn       = var.certificate_arn == null ? aws_acm_certificate.this[0].id : var.certificate_arn
   vault_url             = var.vault_url == null ? join(".", [var.name, var.domain_name]) : var.vault_url
 
-  pillar_extra_config = {
-    for item in var.vault_pillar_extra_config : join("_", [item.type, item.name, "config"]) => jsonencode(item.config)
-  }
+  # extract the env variables and build a map of vars. use this map for templating the pillar
+  pillar_ssm_parameters = { for key, value in var.pillar_ssm_parameters_keys : value => lookup(data.external.env.result, format("TF_VAR_%s", value), "") }
```

`var.pillar_ssm_parameters_keys` is a list, so don't need to use the map syntax... And if the user provides an item in the list, and the item is not in the env result, wouldn't we want to error rather than provide an empty `""` valut?

```suggestion
  pillar_ssm_parameters = { for key in var.pillar_ssm_parameters_keys : key => data.external.env.result[format("TF_VAR_%s", key)] }
```

---

This could use some explanation in the readme. Is terraform looking up the value from SSM? Is salt? For a given value in the list, what is need in the env? What does the user need to do to set the env?

---

We don't need this anymore with the new implementation.